Library/Homebrew/cleanup.rb

```diff
@@ -23,10 +24,27 @@ def self.disk_cleanup_size
       @disk_cleanup_size
     end
 
+    def self.add_unremovable_keg(keg)
+      @unremovable_kegs << keg
+    end
+
+    def self.unremovable_kegs
+      @unremovable_kegs
+    end
```

Comment:
I went with the first option based on not finding any instances of attr_reader through HBC where the class only has class methods.